util/reggen/ip_block.py

```diff
@@ -348,3 +348,21 @@ def get_signal_by_name_as_dict(self, name: str) -> Dict[str, object]:
         else:
             raise ValueError("Signal {} does not exist in IP block {}"
                              .format(name, self.name))
+
+    def is_reg_shadowed(self) -> bool:
+        '''Return boolean indication whether reg block contains shadowed reigsters'''
+
+        if len(self.reg_blocks) == 1 and None in self.reg_blocks:
+            return self.reg_blocks[None].shadowed
```

Comment:
I don't think this code is actually needed: in this case, the `for` loop below will have the same effect, I think?

Reply:
let me double check, because i thought i ran into syntax issues when i try to iterate directly and there was a "none" entry.